chia/full_node/weight_proof.py

```diff
@@ -880,7 +880,7 @@ def _map_sub_epoch_summaries(
         if idx < len(sub_epoch_data) - 1:
             delta = 0
             if idx > 0:
-                delta = sub_epoch_data[idx].num_blocks_overflow
+                delta = data.num_blocks_overflow
```

Comment:
caveat on this bit since we are iterating pairwise with the `[idx + 1]` indexing below.  would be nice to clean up that iteration so what we're doing is described right at the `for`.

i'll just let this do the thing ruff wants here since both before and after we both use `data` and index using `idx`.